src/scope.ml

```diff
@@ -118,6 +118,95 @@ type mode =
+  match t.elt with
+  | P_Prod(args, _) -> List.map (fun x -> match x with (_,_,i) -> i) args
+  | P_Abst(args,_)  -> 
+      begin
+        (* print_string "debug : P_Abst found! \n"; *)
+        List.map (fun x -> match x with (_,_,i) -> i) args
+      end
+  | P_Vari(id, imp) -> if imp=FullyExplicit then [] 
+    else
+    begin
+      (* print_string "debug : P_Vari found! \n"; *)
+      match (StrMap.find_opt (snd id.elt) ss.in_scope) with
+      | Some(res) -> let implicits = (fst res).sym_implicits in
+                      begin
+                        (* print_string "debug : found it in in_scope ! \n"; *)
+                        (* print_string (boolList_to_string implicits); *)
+                        implicits
+                      end
+      | None      -> 
+        (match (StrMap.find_opt (snd id.elt) ss.builtins) with
+        | Some(res) -> (* print_string "debug : found it in builtins ! \n"; *) 
+                       (fst res).sym_implicits
+        | None -> (* print_string "debug : not found ! \n"; *)
+                     [])
+    end
+  | P_Patt(_,_)     -> (* print_string "debug : Pattern ! \n"; *)
+                       [] (* TO DO : see if we want to have implicits for patterns as well, but I don't think so *)
+  | _               -> []
+
+(** [addImplicitArgs args param] builds the full arguments list, using the given arguments 
+    [args] and with the insertion of "_" for the implicit arguments, following the information 
+    provided by the formal parameters [param] *)
+let addImplicitArgs : bool list -> p_term list -> p_term list = fun param args ->
+  let rec addImplicitArgs_aux : bool list -> p_term list -> p_term list -> p_term list = 
+    fun param args fullArgs ->
+    match (param,args) with
+    | (isImplicit::ps, arg1::args') -> 
+      if isImplicit then
+        begin
+          (* print_string "debug : I found an implicit ! \n"; *)
+          addImplicitArgs_aux ps args ((none P_Wild)::fullArgs) (* We did not use the first concrete argument *)
+        end
+      else
+        begin
+          (* print_string "debug : it's not an implicit. \n"; *)
+          addImplicitArgs_aux ps args' (arg1::fullArgs)         (* We did use the first concrete argument *)
+        end
+    | ([], _::_) -> (* print_string "debug : not enough formal parameters. \n"; *)
+                    (List.rev args) @ fullArgs
+    | (_, _)     -> fullArgs
+  in
+    (* If we don't have implicitness information (as for a fully explicit identifier like "@f"), 
+       we directly put only the given concrete args *) 
+    if (param = []) then args 
+    (* We reverse the arguments as they've been built in the opposite order *)
+    else List.rev (addImplicitArgs_aux param args [])
+
+(** [add_arg_tbox t args] builds the application of t to a list
+    of arguments [args] where everything is a tbox *)
+let add_arg_tbox : tbox -> tbox list -> tbox = fun t args ->
```

Comment:
Should be in basics.ml.

Reply:
Well, it should even be replaced by `List.fold_left _Appl t args`...

Reply:
There are similar functions in menhir_parser.mly (get_args, add_args) that should be moved to syntax.ml.

Reply:
I assume you are talking about `splitFunArgs`, there is no good reason to even define a function for adding boxed arguments to a boxed function. But indeed, if `splitFunArgs` is indeed required, I would agree that it should be defined in `src/syntax.ml`. And called `get_args` for consistency with `Basics.get_args` (which works on `Terms.term`).